src/for_2D_build/meshes/mesh_iterators.hpp

```diff
@@ -84,4 +84,46 @@ void mesh_parallel_for(const MeshRange &mesh_range, const LocalFunction &local_f
         ap);
 }
 //=================================================================================================//
+template <typename LocalFunction, typename... Args>
+void mesh_split_parallel_for(const MeshRange &mesh_range, const Arrayi &stride, const LocalFunction &local_function, Args &&...args)
+{
+    // forward sweeping
+    for (int m = 0; m < stride[0]; m++)
+        for (int n = 0; n < stride[1]; n++)
+        {
+            parallel_for(
```

Comment:
I haven't found `parallel_for(block, ...)` with a stride, so I used mod to skip the indices.
Alternatively, I can use two 1D loops `parallel(index begin, index end, index step, ...)` to do this. Which one do you prefer?

Reply:
Use two 1d `parallel_for` for now